pkg/artifactory/resource_artifactory_permission_target.go

```diff
@@ -352,6 +352,9 @@ func resourcePermissionTargetRead(d *schema.ResourceData, m interface{}) error {
 	}
 
 	permissionTarget, resp, err := c.V2.Security.GetPermissionTarget(context.Background(), d.Id())
+	if resp == nil {
```

Comment:
Check `err` first. If you check error first you probably never need to check if `resp` is `nil`.